ymer/src/main/java/com/avanza/ymer/MongoDocumentCollection.java

```diff
@@ -89,6 +95,11 @@ public Document findById(Object id) {
 		return toStream(collection.find(query.getQueryObject()));
 	}
 
+	@Override
+	public Stream<List<Document>> findByQuery(Query query, int batchSize) {
+		return StreamUtils.buffer(toStream(collection.find(query.getQueryObject()).batchSize(batchSize)), batchSize);
```

Comment:
is this method really needed? Seems like the batch size in mongo always tries to fill 16M of data. Seems you can lower the batch size, but not increase it.
https://docs.mongodb.com/manual/tutorial/iterate-a-cursor/#cursor-batches

Reply:
> New in version 3.4: Operations of type find(), aggregate(), listIndexes, and listCollections return a maximum of 16 megabytes per batch. batchSize() can enforce a smaller limit, but not a larger one.

I read that to mean that you can enforce a smaller limit _than 16 megabytes_, not a larger one than that.

> find() and aggregate() operations have an initial batch size of 101 documents by default. Subsequent getMore operations issued against the resulting cursor have no default batch size, so they are limited only by the 16 megabyte message size.

I read this to mean that _if you do not set a batch size_, the first call will fetch 101 documents while subsequent calls fetch 16 megabytes, but you can override this by setting a batch size. 

I think this means that if your specified batch size results in data that totals more than 16 megabytes, the 16 megabyte limit will be used.

Also, I wanted to do the buffering of the resulting stream in this method, to match the batch size.